server/server.go

```diff
@@ -0,0 +1,133 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/minhaz11/cache"
+	"go.uber.org/zap"
+)
+
+const (
+	maxWorkers = 100
+	port       = ":6369"
+)
+
+type Server struct {
+	cache    *cache.Cache
+	Logger   *zap.SugaredLogger
+	listener net.Listener
+	ctx      context.Context
+	cancel   context.CancelFunc
+}
+
+func NewServer() (*Server, error) {
+	logger, err := zap.NewProduction()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create logger: %w", err)
+	}
+	sugar := logger.Sugar()
+
+	c, err := cache.NewCache("cache.json")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create cache: %w", err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	return &Server{
+		cache:  c,
+		Logger: sugar,
+		ctx:    ctx,
+		cancel: cancel,
+	}, nil
+}
+
+func (s *Server) startListener() error {
+	ln, err := net.Listen("tcp", port)
+	if err != nil {
+		return fmt.Errorf("error starting server: %w", err)
+	}
+	s.listener = ln
+	s.Logger.Infof("Cache server is listening on port %s...", port)
+	return nil
+}
+
+func (s *Server) setupWorkerPool(connectionChan chan net.Conn) *sync.WaitGroup {
+	var wg sync.WaitGroup
+
+	for i := 0; i < maxWorkers; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for {
+				select {
+				case <-s.ctx.Done():
+					return
+				case conn, ok := <-connectionChan:
+					if !ok {
+						return
+					}
+
+					_ = conn.SetReadDeadline(time.Now().Add(30 * time.Second))
+					_ = conn.SetWriteDeadline(time.Now().Add(30 * time.Second))
+
```

Comment:
Unhandled error, if there is an error, you are still proceeding with handle connection.